File: app/src/main/java/com/main/climbingdiary/models/RouteSort.java
package com.main.climbingdiary.models;

import android.util.Log;

public enum RouteSort {
    //default sort is by date
    LEVEL,AREA,DATE;
    public String typeToString(){
        return this.toString().toLowerCase();
    }

    public static RouteSort stringToSportType(String type){
        RouteSort value = null;
        try {
            value = RouteSort.valueOf(type.toUpperCase());
        }catch(Exception ex){
            Log.d("Exception in MenuValues stringToSportType() ",ex.getLocalizedMessage());
        }
        return value;
    }
}

File: app/src/main/java/com/main/climbingdiary/models/RouteSort.kt
package com.main.climbingdiary.models

import android.util.Log
import java.util.*

enum class RouteSort {
    LEVEL, AREA, DATE;

    companion object {
        @JvmStatic
        fun stringToSportType(type: String?): RouteSort? {
            var value: RouteSort? = null
            try {
                value = valueOf(type!!.toUpperCase(Locale.ROOT))
            } catch (ex: Exception) {
                Log.d("Exception in MenuValues stringToSportType() ", ex.localizedMessage)
            }
            return value
        }
    }

    fun typeToString(): String {
        return this.toString().toLowerCase(Locale.ROOT)
    }
}